### PR TITLE
update profile related docs

### PR DESCRIPTION
**Pull request by kunmingg**
- include design overview
- include Prerequisite and supported platforms
- include customization guide through plugin interface

fix: https://github.com/kubeflow/website/issues/1124

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1603)

### CODE REVIEW

File: content/docs/other-guides/multi-user-overview.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ weight = 50
 
 {{% stable-status %}}
 
-As of v0.6, Kubeflow supports for multi-user isolation of user-created
+As of v0.6, Kubeflow added multi-user support which apply access control over namespaces / user-created
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: Is "namespaces / user-created " a typo? I don't really understand what the "/" means

File: content/docs/other-guides/multi-user-overview.md [hunk 2]
```diff
@@ -17,12 +17,66 @@ Note that the isolation support in Kubeflow doesn't provide any hard security
 guarantees against malicious attempts by users to infiltrate other user's
 profiles.
 
+## Design overview
+
+Kubeflow multi-tenancy implementation currently follows:
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: The markdown is not correctly rendering as a list.
https://deploy-preview-1603--competent-brattain-de2d6d.netlify.com/docs/other-guides/multi-user-overview/

File: content/docs/other-guides/multi-user-overview.md [hunk 3]
```diff
@@ -17,12 +17,66 @@ Note that the isolation support in Kubeflow doesn't provide any hard security
...
+- Kubeflow UI need to be served behind an identity aware proxy, the identity aware proxy and k8s
+master should share the same identity management.
+  * On GCP we use [GKE](https://cloud.google.com/kubernetes-engine) + [IAP](https://cloud.google.com/iap/docs/concepts-overview)
+  * An alternative is [Dex](https://github.com/dexidp/dex) + LDAP / Active Directory
+
+#### supproted platform
+* kubeflow multi-tenancy is enabled by default if you [deploy kuebflow on GCP with IAP](/docs/gke/deploy)
+* Not on GCP? [deploy to your existing cluster](/docs/started/k8s/kfctl-existing-arrikto/)
+
+### Extend profile to integrate with your platform
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: This seems like developer oriented documentation I'm not sure it belongs here.

File: content/docs/other-guides/multi-user-overview.md [hunk 4]
```diff
@@ -17,12 +17,66 @@ Note that the isolation support in Kubeflow doesn't provide any hard security
...
+
+#### Prerequisite
+- Kubeflow use [Istio](https://istio.io/) to apply access control over in-cluster traffics.
+- Kubeflow profile controller need `Cluster admin` permission.
+- Kubeflow UI need to be served behind an identity aware proxy, the identity aware proxy and k8s
+master should share the same identity management.
+  * On GCP we use [GKE](https://cloud.google.com/kubernetes-engine) + [IAP](https://cloud.google.com/iap/docs/concepts-overview)
+  * An alternative is [Dex](https://github.com/dexidp/dex) + LDAP / Active Directory
+
+#### supproted platform
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: Typo "supported" platform

File: content/docs/other-guides/multi-user-overview.md [hunk 5]
```diff
@@ -17,12 +17,66 @@ Note that the isolation support in Kubeflow doesn't provide any hard security
...
+  class="mt-3 mb-3 border border-info rounded">
+
+### Prerequisite and supported platforms
+
+#### Prerequisite
+- Kubeflow use [Istio](https://istio.io/) to apply access control over in-cluster traffics.
+- Kubeflow profile controller need `Cluster admin` permission.
+- Kubeflow UI need to be served behind an identity aware proxy, the identity aware proxy and k8s
+master should share the same identity management.
+  * On GCP we use [GKE](https://cloud.google.com/kubernetes-engine) + [IAP](https://cloud.google.com/iap/docs/concepts-overview)
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: Do we need to be GCP specific here?

Comment (kunmingg) [reply content/docs/other-guides/multi-user-overview.md]: GKE + IAP is only available on GCP, the alternative is Dex?

**Comment by jlewi** [comment]
Thanks @kunmingg 

I have a couple high level questions that it might be good to get consensus on.

* What do we want this doc do be about? Do we want it to be "here's how you use profiles? Or do we want more conceptual overview of IAM in Kubeflow?
* How do we want to mix generic information with Cloud specific information?

It looks like the original issue #1124 was about profiles. Should we try to answer some very specific questions about profiles?

The initial questions that came to my mind were

* What are Profiles?
* What do Profiles do?
* How do I use profiles with kubectl?
   * What permissions does a user need to be able to create profiles?
* How do I use profiles with the Kubeflow UI?
    * What permissions are needed to be able to access the UI?

Should we create a page under components of Kubeflow dedicated to profiles and move most of this information there?
 
What would we do about the existing multi-user-overview page? Should this become more of a conceptual overview of IAM in Kubeflow? 
   * Should we move this into the Reference Section
   * IAM in Kubeflow is a fairly meaty topic. How much should we aim to cover in this PR?

/cc @sarahmaddox @krishnadurai @yanniszark

**Comment by sarahmaddox** [comment]
/assign
/lgtm

Thanks @kunmingg this is an good addition.

There are a few small grammatical/style points that we can ask a tech writer to look at during the doc sprint.

@jlewi I think you've asked some excellent questions. This page has grown since it was first created, and is now a mix of conceptual overviews, usage information, and tech prerequisites. It could do with splitting up. I already have a task marked for possible tackling by a tech writer during the doc sprint, to look at this page as a whole. (I haven't yet created the issue - it's currently in a spreadsheet of similar tasks for the doc sprint.) There's a bit about profiles, for example, in the [usage overview](https://deploy-preview-1603--competent-brattain-de2d6d.netlify.com/docs/other-guides/multi-user-overview/#usage-overview) section.

How about we merge this PR, and I'll refer to the comments on this PR when I create the doc sprint issue?

File: content/docs/other-guides/multi-user-overview.md (hunk 5 above)
Comment (yanniszark) [reply content/docs/other-guides/multi-user-overview.md]: Let's also mention that we use Dex for on-prem installations.
Similar to what you have written for GCP:
```
For on-prem installations, we make use of Dex, a flexible OIDC provider.
```

**Comment by sarahmaddox** [comment]
@kunmingg I have a suggestion: Will you be able to finalize this PR on Monday to get a minimum viable doc update merged? Then you can work with a tech writer at the doc sprint to split out the doc into multiple pages, and polish the content. We have a number of tech writers taking part in the doc sprint. It's an ideal time to get this done.

**Comment by kunmingg** [comment]
@sarahmaddox 
@richardsliu 
Can we have this one merged and refactor them in follow up PRs?

**Pull request merged**